tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+                scopes = ArrayPool<Model.Scope>.Shared.Rent(scopesBufferLength);
+                int methodsScopeIndex = 0;
+                int nestedClassesScopeIndex = 0;
+                if (methods.Count > 0)
+                {
+                    PopulateMethodScopes(type, methods, scopes, ref methodsScopeIndex);
+                }
+
+                nestedClassesScopeIndex = methodsScopeIndex;
+                if (nestedTypes.Length > 0)
+                {
+                    PopulateNestedNotCompileGeneratedClassScope(nestedTypes, scopes, ref nestedClassesScopeIndex);
+                }
+
+                var allScopes = nestedClassesScopeIndex == 0 ? null : new Model.Scope[nestedClassesScopeIndex];
+
+                if (methodsScopeIndex > 0)
+                {
+                    Array.Copy(scopes!, 0, allScopes!, 0, methodsScopeIndex);
+                }
+
+                if ((nestedClassesScopeIndex - methodsScopeIndex) > 0)
+                {
+                    int destIndex = methodsScopeIndex == 0 ? 0 : methodsScopeIndex;
+                    Array.Copy(scopes!, 0, allScopes!, destIndex, nestedClassesScopeIndex - methodsScopeIndex);
+                }
+
+                var classLanguageSpecifics = GetClassLanguageSpecifics(type);
+                var linesAndSource = GetClassSourcePdbInfo(allScopes);
+                classScope = new Model.Scope
+                {
+                    Name = typeDefinitionHandle.FullName(MetadataReader),
+                    ScopeType = SymbolType.Class,
+                    Symbols = fieldSymbols,
+                    Scopes = allScopes,
+                    StartLine = linesAndSource.StartLine,
+                    EndLine = linesAndSource.EndLine,
+                    SourceFile = linesAndSource.Path,
+                    LanguageSpecifics = classLanguageSpecifics
+                };
+            }
+            catch (Exception e)
+            {
+                string? typeName = null;
+                try
+                {
+                    typeName = typeDefinitionHandle.FullName(MetadataReader);
+                }
+                catch
+                {
+                    // ignored
+                }
+
+                Log.Warning(e, "Error while trying to extract symbol info for type {Type}", typeName ?? "NA");
+                return false;
+            }
+            finally
+            {
+                if (scopes != null)
+                {
+                    ArrayPool<Model.Scope>.Shared.Return(scopes);
+                }
+            }
+
+            return true;
+        }
+
+        private SourcePdbInfo GetClassSourcePdbInfo(Model.Scope[]? allScopes)
+        {
+            int classStartLine = int.MaxValue;
+            int classEndLine = 0;
+            string? classSourceFile = null;
+            if (allScopes == null)
+            {
+                return new SourcePdbInfo { StartLine = classStartLine, EndLine = classEndLine, Path = classSourceFile };
```

Comment:
In case `allScopes` is null, do we want the `StartLine` to be `int.MaxValue` and `EndLine` 0? Seems incorrect.
What about placing `UnknownMethodStartLine` (0) and `UnknownMethodEndLine` (0) respectively?

Reply:
Fixed in f74ef445ed4a6220e0849d0be2ba5866e48a3fa5